### PR TITLE
DwC-A Media Extension File fix

### PR DESCRIPTION
**Pull request by egbot**
- Coordinate field name matching within media extension file. Mismatch of character case usage was causing data updates within array to fail. While the field case used within the code appears to be not standardized (e.g. sometime camalCase and other PascalCase), it is purposely structured to match what is used within the Audiomedia Core standard (aka Audubon Core), which matches terms based on inheritance of different parent standards (e.g. Dublin Core).  

Addresses issue: https://github.com/Symbiota/Symbiota/issues/3296 

# Pull Request Checklist:
# Pre-Approval

- [x] There is a description section in the pull request that details what the proposed changes do. It can be very brief if need be, but it ought to exist
- [ ] Hotfixes should be branched off of the `master` branch and PR'd using the **squash & merge** option into the `Hotfix-x.x.x` branch.
- [ ] Features and backlog bugs should be merged into the `Development` branch, **NOT** `master`
- [ ] All new text is preferably internationalized (i.e., no end-user-visible text is hard-coded on the PHP pages), and the [spreadsheet tracking internationalizations](https://docs.google.com/spreadsheets/d/133fps9w2pUCEjUA6IGCcQotk7dn9KvepMXJ2IWUZsE8/edit?usp=sharing) has been updated either with a new row or with checkmarks to existing rows
- [x] There are no linter errors
- [ ] New features have responsive design (i.e., look aesthetically pleasing both full screen and with small or mobile screens)
- [x] [Symbiota coding standards](https://docs.google.com/document/d/1-FwCZP5Zu4f-bPwsKeVVsZErytALOJyA2szjbfSUjmc/edit?usp=sharing) have been followed
- [ ] If any files have been reformatted (e.g., by an autoformatter), the reformat is its own, separate commit in the PR
- [x] Comment which GitHub issue(s), if any does this PR address
- [ ] If this PR makes any changes that would require additional configuration of any Symbiota portals outside of the files tracked in this repository, make sure that those changes are detailed in [this document](https://docs.google.com/document/d/1T7xbXEf2bjjm-PMrlXpUBa69aTMAIROPXVqJqa2ow_I/edit?usp=sharing)

# Post-Approval

- [ ] It is the code author's responsibility to merge their own pull request after it has been approved
- [ ] If this PR represents a merge into the `Development` branch, remember to use the **squash & merge** option
- [ ] If this PR represents a merge into the `hotfix` branch, remember to use the **squash & merge** option
- [ ] If this PR represents a merge from the `Development` branch into the master branch, remember to use the **merge** option
- [ ] If this PR represents a merge from the `hotfix` branch into the `master` branch use the **merge** option (not squashed)
  - [ ] a subsequent PR from `master` into `Development` should be made with the **merge** option (i.e., no squash)
  - [ ] **Immediately** delete the `hotfix` branch and create a new `hotfix` branch
  - [ ] increment the Symbiota version number in the symbase.php file and commit to the `hotfix` branch
- [ ] If the dev team has agreed that this PR represents the last PR going into the `Development` branch before a tagged release (i.e., before an imminent merge into the master branch), make sure to notify the team and [lock the `Development` branch](https://github.com/BioKIC/Symbiota/settings/branches) to prevent accidental merges while QA takes place. Follow the release protocol [here](https://github.com/BioKIC/Symbiota/blob/master/docs/release-protocol.md)
- [ ] Don't forget to delete your feature branch upon merge. Ignore this step as required

Thanks for contributing and keeping it clean!

### CODE REVIEW

File: classes/DwcArchiverMedia.php [hunk 1]
```diff
@@ -79,8 +79,9 @@ private function trimBySchemaType($dataArr){
 	private function setSql(){
 		if($this->fieldArr){
 			$sqlFrag = '';
-			foreach($this->fieldArr['fields'] as $colName){
+			foreach($this->fieldArr['fields'] as $fieldName =>$colName){
 				if($colName) $sqlFrag .= ', ' . $colName;
+				else  $sqlFrag .= ', "" AS ' . $fieldName;
```
Comment (Atticus29) [inline classes/DwcArchiverMedia.php:84]: I'm confused about this line:
```
If the value in the array is missing,
use "AS" . $key instead
```
?

File: classes/DwcArchiverMedia.php [hunk 2]
```diff
@@ -125,34 +126,34 @@ public function writeOutMediaData($exportID, $collArr, $serverDomain){
 						}
 						$collid = $r['collid'];
 						unset($r['collid']);
-						$r['rights'] = $collArr[$collid]['rights'];
-						$r['owner'] = $collArr[$collid]['rightsholder'];
-						$r['webstatement'] = $collArr[$collid]['accessrights'];
+						if(!$r['rights']) $r['rights'] = $collArr[$collid]['rights'];
```
Comment (Atticus29) [inline classes/DwcArchiverMedia.php:129]: Maybe better to use array_key_exists here instead? Otherwise, I think we'll see development warnings.

**Review by Atticus29 — APPROVED** [review]
Definitely seems to be working now. I have a few questions (see comments), but none are very likely to be blocking.

Top: before. Bottom: after: 

Image

File: classes/DwcArchiverMedia.php (hunk 1 above)
Comment (egbot) [reply classes/DwcArchiverMedia.php:84]: We still need this field included within the SQL, even if the output is nothing, otherwise the columns won't match the output. Thus, if the SQL fragment is not defined in the fieldArr, then it would add something like "" as WebStatement, which just outputs empty strings for that field. These empty strings are then populated in the code using other content inserted into the output array.

File: classes/DwcArchiverMedia.php (hunk 2 above)
Comment (egbot) [reply classes/DwcArchiverMedia.php:129]: It always exists within the output, thus array_key_exists would always be true, but I need to test if there is a non-empty value. empty($r['rights']) might be better, since it would test true if existing and has value, but not really needed since it will always be present within the output array.
